### PR TITLE
Baseline measurement with SIMON

### PR DESCRIPTION
**Pull request by e13h**
Provide a baseline measurement using [SIMON](https://github.com/NewKnowledge/simon).
In this PR:
 * SIMON is included as a git submodule (I was unable to figure out how to simply install SIMON using `pip install` inside a `conda` environment configured with `tensorflow`).
 * `README.md` is updated with a guide on how to run the baseline in a `conda` environment
 * The script `simon.py` is added to run the baseline

### CODE REVIEW

File: simon.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+    encoder = Encoder(categories=categories)
+    encoder.process(raw_data, MAX_CELLS)
+    X, y = encoder.encode_data(raw_data, header, MAX_LEN)
+    encoder_max_cells = encoder.cur_max_cells
+    category_count = y.shape[1] # need to know number of fixed categories to create model
+    print('Number of fixed categories is: {}'.format(category_count))
+
+    classifier = Simon(encoder=encoder)
+    data = classifier.setup_test_sets(X, y)
```
Comment (bjschoenfeld) [inline simon.py:39]: @evanphilipsmith We need to make a moderate change here. The splitting of data, whether train/test split or CV, should be performed consistently when comparing SIMON to our model or any other model. We need to make sure that the splits are the same so that the score compared are the same. Can you look making this change?

Comment (e13h) [reply simon.py:39]: @bjschoenfeld 
SIMON: 60% train, 30% validation set, 10% test set (code [here](https://github.com/NewKnowledge/simon/blob/4123c99004eda92a35ed404bc72274d269bbe7b6/Simon/Simon.py#L101-L103) for reference)
BYU profiler: 67% train, 33% test (code [here](https://github.com/byu-dml/d3m-profiler/blob/30779327fffcb546797c44d30013791bd0b89d4d/example.py#L72) for reference)

I could pretty easily change the way the data is split for SIMON's model to mimic ours, but it seems like the Keras model has some callbacks that are specifically looking at the validation set loss, (see their training function [here](https://github.com/NewKnowledge/simon/blob/4123c99004eda92a35ed404bc72274d269bbe7b6/Simon/Simon.py#L306-L323)). The model crashes when I pass it empty validation sets.

I would assume that early stopping is important to the model's performance. Should I focus my efforts on changing our model to implement a validation set? Or do you think there is another way to adjust SIMON's test/train/validation set split to resemble ours more closely?

Comment (e13h) [reply simon.py:39]: I was able to use a 67% training set and 33% test set by removing the early stopping callbacks from the Keras model. The results from both versions are below are:
```
                                            F1 macro     F1 macro
                                            (zero-div    (zero-div 
Validation Set?     Accuracy   F1 micro     returns 0)   returns 1)     F1 weighted
            no        0.9928      0.9681        0.4113       0.4113          0.9695
           yes        0.9923      0.9636        0.4122       0.6622          0.9652
```

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "simon"]
+	path = simon
+	url = https://github.com/NewKnowledge/simon.git
```
Comment (bjschoenfeld) [inline .gitmodules:3]: Why is simon a submodule? Can we just install it?

File: simon.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    probabilities = model.predict(data.X_test, verbose=1)
+    prediction_indices = probabilities > P_THRESHOLD
+    y_pred = np.zeros(data.y_test.shape)
+    y_pred[prediction_indices] = 1
+    print(f'F1 micro: {f1_score(data.y_test, y_pred, average="micro")}')
+    print(f'F1 macro: {f1_score(data.y_test, y_pred, average="macro")}')
+    print(f'F1 weighted: {f1_score(data.y_test, y_pred, average="weighted")}')
+
+
+def parse_datasets(datasets):
```
Comment (bjschoenfeld) [inline simon.py:70]: d3m has functionality for loading d3m datasets: https://gitlab.com/datadrivendiscovery/d3m/-/blob/devel/d3m/container/dataset.py#L3193

Using this means we don't have to maintain our own dataset parsing code.

**Comment by bjschoenfeld** [comment]
Is this closed in favor of the other PR?

**Comment by e13h** [comment]
> Is this closed in favor of the other PR?

Yes
